### DISCUSSION TITLE
psqldef cannot convert SERIAL family.

### DISCUSSION THREAD

**Discussion by chumaltd** [sqldef/sqldef#628] [Feature requests]
Hi, I just hit an AUTO INCREMENT id conversion issue.

PostgreSQL has syntax sugar  `serial`, `bigserial` just for `CREATE TABLE`.  
psqldef issues `ALTER TABLE ... ALTER COLUMN ... TYPE bigserial`, and fails.

For actual example, Rails 5.1 once converted default `:id` type from `integer` to `bigint`.  
I just solved this `serial` -> `bigserial` conversion with manual `ALTER` listed the last part.

I think `psqldef` parser is sure to handle directly `serial` or `bigserial`, as it can handle no change type senarios quite well.
But it wants to have a extra handler for `SEQUENCE` on later stage.

I think this issue has more couples like `integer` -> `serial` .

This is not urgent for me. They can be solved by manual operations.
I prefer a good design !

## Platform

* OS: Linux
* RDBMS: PostgreSQL 16
* Version:  0.16.10

## --export output

```sql
CREATE TABLE "public"."users" (
    "id" serial NOT NULL
);
```

## Input SQL

```sql
CREATE TABLE "public"."users" (
    "id" bigserial NOT NULL
);
```

## Current output

``` bash
$ cat sqldef.sql2 | psqldef -Upostgres sqldeftest
-- Apply --
ALTER TABLE "public"."users" ALTER COLUMN "id" TYPE bigserial;
2023/11/15 23:00:28 pq: type "bigserial" does not exist
```

## Expected output

For this case, required SQL set is following:

```sql
ALTER TABLE users ALTER COLUMN id TYPE bigint;
ALTER SEQUENCE users_id_seq AS bigint;
```

**Comment by gfx** [comment]
Resolved in https://github.com/sqldef/sqldef/pull/1049
